### PR TITLE
bench: add CAddrMan benchmarks

### PR DESCRIPTION
**Pull request by vasild**
The added benchmarks exercise the public methods Add(), GetAddr(),
Select() and Good().

### CODE REVIEW

**Comment by vasild** [comment]
Extracted from https://github.com/bitcoin/bitcoin/pull/18722 upon [request](https://github.com/bitcoin/bitcoin/pull/18722#issuecomment-617828865).

File: src/bench/addrman.cpp [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+static std::vector<CAddress> g_sources;
+static std::vector<std::vector<CAddress>> g_addresses;
+
+static void CreateAddresses()
+{
+    if (g_sources.size() > 0) { // already created
+        return;
+    }
+
+    FastRandomContext rng(uint256(std::vector<unsigned char>(32, 123)));
```
Comment (MarcoFalke) [inline src/bench/addrman.cpp:26]: nit: I think you can just set 
```suggestion
    FastRandomContext rng(/* deterministic */ true);
```

or is there a reason for the seed?

**Review by maflcko — APPROVED** [review]
ACK a9b957740e3490d87e5ce0b7f1b93ba43bb19764

**Comment by MarcoFalke** [comment]
cc @naumenkogs (you might be interested in this)

**Comment by naumenkogs** [comment]
utACK a9b9577

File: src/bench/addrman.cpp (hunk 1 above)
Comment (vasild) [reply src/bench/addrman.cpp:26]: Right, `/* deterministic */ true` would have been better. I will have this in mind and also change this code if I am touching it again.

There is no need for the chosen seed - it only has to be the same every time to reduce noise in performance tests.

Thanks!

**Pull request merged**